src/main/java/Duke.java

```diff
@@ -1,10 +1,78 @@
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Represents the Duke Chatbot.
+ * Allows users to store a list of Tasks, consisting of three types,
+ * namely Todos, Events and Deadlines. Users input, change and delete
+ * such items via text commands to the bot.
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Builds an instance of the Duke Chatbot.
+     */
+    public Duke() {
+        ui = new Ui();
+        storage = new Storage();
+        tasks = new TaskList(storage.load());
+    }
+
+    /**
+     * Runs the chatbot instance.
+     * Chatbot will begin listening for commands with this method.
+     */
+    public void run() {
+        try {
+            ui.intro();
+            Scanner input = new Scanner(System.in);
+
+            // Uses user input text and acts accordingly.
+            loop:
```

Comment:
I have never seen this syntax before? Is it an meant to be commented out?

Reply:
No it is not! It was meant to break out of the outer while loop from within the switch case, as a normal `break` would only break out of the switch case.

I used this to solve the issue I had: https://stackoverflow.com/questions/22823395/java-how-can-i-break-a-while-loop-under-a-switch-statement

Reply:
Wow, thanks for sharing man